<a href="https://colab.research.google.com/github/Matheus-Barbosa-Melo-dos-Santos/analise-amazon/blob/main/notebook_de_analise.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Projetos de Análise de Dados.

## Escopo.
Usando as ferramentas do google workspace, colab e aistudio, farei uma
análise de uma base de dados, onde será entregue em formas de relatórios e afins.

### Base de dados:
A base desse projeto foi retirada do [Kaggle](https://www.kaggle.com/), onde a [base analisada](https://www.kaggle.com/datasets/ashishkumarak/amazon-shopping-reviews-daily-updated/data) possuindo uma fonte de reviews.

In [1]:
# Importando as bibliotecas dash e da base de dados.

!pip install dash
import pandas as pd
import dash
import dash_html_components as html
import dash_core_components as dcc
from dash.dependencies import Input, Output

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.5/7.5 MB 19.3 MB/s eta 0:00:00


<ipython-input-1-ecd5cf7e23ef>:6: UserWarning: 
The dash_html_components package is deprecated. Please replace
`import dash_html_components as html` with `from dash import html`
  import dash_html_components as html
<ipython-input-1-ecd5cf7e23ef>:7: UserWarning: 
The dash_core_components package is deprecated. Please replace
`import dash_core_components as dcc` with `from dash import dcc`
  import dash_core_components as dcc


In [2]:
# Acessando a base de dados.

df_amazon = pd.read_csv('/content/amazon_reviews.csv')
df_amazon.head()

reviewId        userName  \
0  a37d5804-60a3-4b02-96dd-248ea794e0a0           Ahmed   
1  fa14ee30-3621-4f4a-8dec-f3fc7a884e8e      Rana Nawar   
2  11d2f9f4-71ca-42df-b690-9d7a99bf91f1   valerie musil   
3  374d55af-e4a5-44b6-ab6c-66cada79b0df  Marilyn Slentz   
4  ad246467-0bc3-4679-b193-ac9e18f37fb7       JC Lawery   

                                             content  score  thumbsUpCount  \
0                            Too bad and not working      1              0   
1                                           Very bad      1              0   
2  I love amazon. I have never had a problem with...      5              0   
3  I can't delete this app and it pisses me to no...      1              0   
4  I just recently canceled an order and you not ...      1              0   

  reviewCreatedVersion                   at   appVersion  
0                  NaN  2024-05-28 22:08:26          NaN  
1           28.7.0.100  2024-05-28 21:55:00   28.7.0.100  
2          28.10.0.100  2024-05-28 21:54:18  28.10.0.100  
3          24.10.2.100  2024-05-28 21:53:31  24.10.2.100  
4          28.10.0.100  2024-05-28 21:52:35  28.10.0.100

In [3]:
# Vendo a quantidade de linhas e colunas no dataframe.

df_amazon.shape

(51913, 8)

### 1° vista.

A base de dados possui 51.913 linhas e 8 colunas, onde trazem as reviews do app da amazon na Play Store, com isso, podemos levantar alguns questionamentos.

1.   Quantas reviews postivias e negativas tivemos?
2.   Sobre o que se tratam o review?
3.   Quando foram feitos comentarios?
4.   Sobre qual versão do App eles se tratam?

Porém, antes irei rastrear e analisar alguns possiveis erros e limpar um pouco da planilha, por hora filtrarei a planilha atras de todos os `NaN`.


In [4]:
# Fazendo a tratativa de dados.
# Primeiro, farei uma copia do dataframe para analise eliminaremos os na.

df_amazon_analise = df_amazon.copy()
df_amazon_analise.dropna(inplace=True)
df_amazon_analise.shape

(46868, 8)

In [5]:
# Ficando com 46.868 linhas, abrirei o df.

df_amazon_analise.head()

reviewId        userName  \
1  fa14ee30-3621-4f4a-8dec-f3fc7a884e8e      Rana Nawar   
2  11d2f9f4-71ca-42df-b690-9d7a99bf91f1   valerie musil   
3  374d55af-e4a5-44b6-ab6c-66cada79b0df  Marilyn Slentz   
4  ad246467-0bc3-4679-b193-ac9e18f37fb7       JC Lawery   
5  90e577b7-a79d-4d2a-b0c1-5e4e4405c4bb    Haylee babii   

                                             content  score  thumbsUpCount  \
1                                           Very bad      1              0   
2  I love amazon. I have never had a problem with...      5              0   
3  I can't delete this app and it pisses me to no...      1              0   
4  I just recently canceled an order and you not ...      1              0   
5  I love it. I shop more on here than anywhere else      5              0   

  reviewCreatedVersion                   at   appVersion  
1           28.7.0.100  2024-05-28 21:55:00   28.7.0.100  
2          28.10.0.100  2024-05-28 21:54:18  28.10.0.100  
3          24.10.2.100  2024-05-28 21:53:31  24.10.2.100  
4          28.10.0.100  2024-05-28 21:52:35  28.10.0.100  
5          28.10.0.100  2024-05-28 21:41:10  28.10.0.100

In [6]:
# A paritir de agora, vamos as perguntas.
# 1. Quantas reviews postivias e negativas tivemos?

df_amazon_analise['score'].value_counts()

score
1    21912
5    11122
2     6051
3     4545
4     3238
Name: count, dtype: int64

In [7]:
# Tornando esse dado visual

import plotly.express as px

fig = px.pie(df_amazon_analise, values='score', names='score')
fig.show()

In [29]:
# Certo, vamos usar as datas para uma melhor visualização

df_amazon_analise.rename(columns={'at': 'data'}, inplace=True)
df_amazon_analise.head()

reviewId        userName  \
1  fa14ee30-3621-4f4a-8dec-f3fc7a884e8e      Rana Nawar   
2  11d2f9f4-71ca-42df-b690-9d7a99bf91f1   valerie musil   
3  374d55af-e4a5-44b6-ab6c-66cada79b0df  Marilyn Slentz   
4  ad246467-0bc3-4679-b193-ac9e18f37fb7       JC Lawery   
5  90e577b7-a79d-4d2a-b0c1-5e4e4405c4bb    Haylee babii   

                                             content  score  thumbsUpCount  \
1                                           Very bad      1              0   
2  I love amazon. I have never had a problem with...      5              0   
3  I can't delete this app and it pisses me to no...      1              0   
4  I just recently canceled an order and you not ...      1              0   
5  I love it. I shop more on here than anywhere else      5              0   

  reviewCreatedVersion                data   appVersion  
1           28.7.0.100 2024-05-28 21:55:00   28.7.0.100  
2          28.10.0.100 2024-05-28 21:54:18  28.10.0.100  
3          24.10.2.100 2024-05-28 21:53:31  24.10.2.100  
4          28.10.0.100 2024-05-28 21:52:35  28.10.0.100  
5          28.10.0.100 2024-05-28 21:41:10  28.10.0.100

In [33]:
df_amazon_analise.insert(1, 'year', df_amazon_analise['data'].dt.year)
df_amazon_analise.head()

reviewId  year   ano        userName  \
1  fa14ee30-3621-4f4a-8dec-f3fc7a884e8e  2024  2024      Rana Nawar   
2  11d2f9f4-71ca-42df-b690-9d7a99bf91f1  2024  2024   valerie musil   
3  374d55af-e4a5-44b6-ab6c-66cada79b0df  2024  2024  Marilyn Slentz   
4  ad246467-0bc3-4679-b193-ac9e18f37fb7  2024  2024       JC Lawery   
5  90e577b7-a79d-4d2a-b0c1-5e4e4405c4bb  2024  2024    Haylee babii   

                                             content  score  thumbsUpCount  \
1                                           Very bad      1              0   
2  I love amazon. I have never had a problem with...      5              0   
3  I can't delete this app and it pisses me to no...      1              0   
4  I just recently canceled an order and you not ...      1              0   
5  I love it. I shop more on here than anywhere else      5              0   

  reviewCreatedVersion                data   appVersion  
1           28.7.0.100 2024-05-28 21:55:00   28.7.0.100  
2          28.10.0.100 2024-05-28 21:54:18  28.10.0.100  
3          24.10.2.100 2024-05-28 21:53:31  24.10.2.100  
4          28.10.0.100 2024-05-28 21:52:35  28.10.0.100  
5          28.10.0.100 2024-05-28 21:41:10  28.10.0.100

In [64]:
balance = df_amazon_analise['score'].value_counts()
balance

score
1    21912
5    11122
2     6051
3     4545
4     3238
Name: count, dtype: int64

In [68]:
fig=px.bar(df_amazon_analise, x=balance.index, y=balance.values)
fig.show()

In [69]:
fig = px.histogram(df_amazon_analise, x='year', color='score')
fig.show()

Ok, com isso, tenho uma boa visualização da disposição do data frame e respondemos a 1° pergunta: "1.   Quantas reviews postivias e negativas tivemos?"

Sendo que tivemos:

21912 de uma estrela

6051 de duas estrelas

4545 de três estrelas

3238 de quatro estrelas

11122 de Cinco estrelas

